Firebase/Messaging/FIRMessagingPubSub.m

```diff
@@ -195,16 +199,17 @@ - (void)restorePendingTopicsList {
   GULUserDefaults *defaults = [GULUserDefaults standardUserDefaults];
   NSData *pendingData = [defaults objectForKey:kPendingSubscriptionsListKey];
   FIRMessagingPendingTopicsList *subscriptions;
-  @try {
-    if (pendingData) {
-#pragma clang diagnostic push
-#pragma clang diagnostic ignored "-Wdeprecated-declarations"
-      subscriptions = [NSKeyedUnarchiver unarchiveObjectWithData:pendingData];
-#pragma clang diagnostic pop
+  if (pendingData) {
+    if (@available(macOS 10.13, iOS 11.0, tvOS 11.0, watchOS 4.0, *)) {
+      NSError *error;
+      subscriptions = [NSKeyedUnarchiver unarchivedObjectOfClass:[NSData class] fromData:pendingData error:&error];
+    } else {
+      @try {
+        subscriptions = [NSKeyedUnarchiver unarchiveObjectWithData:pendingData];
+      } @catch (NSException *exception) {
+        // Nothing we can do, just continue as if we don't have pending subscriptions
+      }
```

Comment:
I would suggest using [GULSecureCoding](https://github.com/firebase/firebase-ios-sdk/blob/b7fec9e2b82ef91fe4a2422d2f409cfd58c9b4fe/GoogleUtilities/Environment/Public/GULSecureCoding.h#L24) instead of a custom implementation to reduce code duplication.

Reply:
nice, didn't know there's a shared method already.